tapgarden/custodian.go

```diff
@@ -795,9 +771,72 @@ func (c *Custodian) setReceiveCompleted(event *address.Event,
 		Index: lastProof.InclusionProof.OutputIndex,
 	}
 
-	return c.cfg.AddrBook.CompleteEvent(
+	err = c.cfg.AddrBook.CompleteEvent(
 		ctxt, event, address.StatusCompleted, anchorPoint,
 	)
+	if err != nil {
+		return fmt.Errorf("error completing address book event: %w",
+			err)
+	}
+
+	// Remove event from local cache.
+	c.events.Delete(event.Outpoint)
+
+	return nil
+}
+
+// handleNewProof handles a newly received proof by identifying any related
+// address events and marking them as complete.
+func (c *Custodian) handleNewProof(newProof proof.Blob) error {
+	// Obtain proof file from a single proof blob or a proof file blob.
+	var (
+		file             *proof.File
+		proofFileVersion = proof.V0
+	)
+
+	switch {
+	case proof.IsProofFile(newProof):
```

Comment:
Why is this logic needed? I think we'll only ever _not_ get a file once https://github.com/lightninglabs/taproot-assets/pull/726 is in, which already adds this code but in a slightly different way.

Reply:
I've replaced this logic with a check to make sure that the proof blob is actually a file.

Reply:
I've just realised you used a negative "only ever _not_ get a file".

Whatever we're supposed to get at this point, I think we should add a check for it. Note that the blob to file conversion was there already, I didn't add that in this PR. I think we should check each time before we interpret a proof blob as a file or single proof.

I'll check for a single proof if https://github.com/lightninglabs/taproot-assets/pull/726 gets in first.

 I think it would make sense if the `c.proofSubscription` had a proper proof type and not just `Blob`. Or at least `SingleProofBlob` and `FileBlob`. Maybe one day a `VerifiedProof` type.